tls/s2n_handshake.h

```diff
@@ -120,11 +120,12 @@ struct s2n_handshake {
 /* Has the handshake been negotiated yet? */
 #define INITIAL                     0x00
 #define NEGOTIATED                  0x01
+#define IS_NEGOTIATED( type ) ( (type) & NEGOTIATED )
 
 /* Handshake is a full handshake  */
 #define FULL_HANDSHAKE              0x02
 #define IS_FULL_HANDSHAKE( type )   ( (type) & FULL_HANDSHAKE )
-#define IS_RESUMPTION_HANDSHAKE( type ) ( !IS_FULL_HANDSHAKE( (type) ) )
+#define IS_RESUMPTION_HANDSHAKE( type ) ( !IS_FULL_HANDSHAKE( (type) ) && IS_NEGOTIATED ( (type) ) )
 
```

Comment:
Is this the right fix?  A resumption handshake type isn’t necessarily successful.

Should is_session_resumed be fixed instead?

Reply:
`IS_NEGOTIATED` doesn't quite imply the handshake was successful. It just means the handshake type has been decided, which happens after the client hello is processed on the server side. Maybe `IS_DECIDED` would be more clear for that flag. https://github.com/awslabs/s2n/blob/master/tls/s2n_handshake_io.c#L310


I don't have a strong feeling here but I slightly prefer the current approach because it will prevent a similar issue with code that uses `IS_RESUMPTION_HANDSHAKE` in the future.